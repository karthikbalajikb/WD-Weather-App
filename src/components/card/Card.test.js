import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import WDCard from './Card';

describe('Component: WDCard', () => {
  const WeatherReport = {
    temp_max: 23,
    description: 'test',
    humidity: 100,
    wind: 20,
    icon: 'http://openweathermap.org/img/w/01n.png',
  };

  it('Renders without crashing', () => {
    shallow(<WDCard WeatherReport={WeatherReport} />);
  });

  it('Snapshot testing', () => {
    const tree = renderer.create(<WDCard WeatherReport={WeatherReport} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
