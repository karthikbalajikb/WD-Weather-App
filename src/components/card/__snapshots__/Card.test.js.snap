// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: WDCard Snapshot testing 1`] = `
<div>
  <div
    className="MuiPaper-root-8 MuiPaper-shadow2-12 MuiPaper-rounded-9 WDCard-card-1"
  >
    <div
      className="WDCard-details-2"
    >
      <div
        className="MuiCardContent-root-35 WDCard-content-3"
      >
        <h1
          className="MuiTypography-root-36 MuiTypography-display4-37 WDCard-font-7"
        >
          23
          <span>
            Â°C
          </span>
        </h1>
        <h1
          className="MuiTypography-root-36 MuiTypography-headline-41 MuiTypography-colorTextSecondary-58"
        >
          <span>
             
            test
             
          </span>
        </h1>
        <h3
          className="MuiTypography-root-36 MuiTypography-subheading-43 MuiTypography-colorTextSecondary-58"
        >
          <span>
            Humidity : 
            100
            %
          </span>
        </h3>
        <h3
          className="MuiTypography-root-36 MuiTypography-subheading-43 MuiTypography-colorTextSecondary-58"
        >
          <span>
            Wind 
            20
             m/s 
          </span>
        </h3>
      </div>
      <div
        className="WDCard-controls-5"
      />
    </div>
    <div
      className="MuiCardMedia-root-60 WDCard-cover-4"
      src={undefined}
      style={
        Object {
          "backgroundImage": "url(http://openweathermap.org/img/w/01n.png)",
        }
      }
    />
  </div>
</div>
`;
